src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -38,145 +39,150 @@
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Strings.isNullOrEmpty;
 import static java.util.Objects.requireNonNull;
 
 public class SlackEventNotification implements EventNotification {
 
-	public interface Factory extends EventNotification.Factory {
-		@Override
+    private static final Logger LOG = LoggerFactory.getLogger(SlackEventNotification.class);
+    private final EventNotificationService notificationCallbackService;
+    private final Engine templateEngine;
+    private final NotificationService notificationService;
+    private final ObjectMapper objectMapper;
+    private final NodeId nodeId;
+    private final SlackClient slackClient;
+
+    @Inject
+    public SlackEventNotification(EventNotificationService notificationCallbackService,
+                                  ObjectMapper objectMapper,
+                                  Engine templateEngine,
+                                  NotificationService notificationService,
+                                  NodeId nodeId, SlackClient slackClient) {
+        this.notificationCallbackService = notificationCallbackService;
+        this.objectMapper = requireNonNull(objectMapper);
+        this.templateEngine = requireNonNull(templateEngine);
+        this.notificationService = requireNonNull(notificationService);
+        this.nodeId = requireNonNull(nodeId);
+        this.slackClient = requireNonNull(slackClient);
+    }
+
+    /**
+     * @param ctx
+     * @throws EventNotificationException is thrown when send fails
+     */
+    @Override
+    public void execute(EventNotificationContext ctx) throws EventNotificationException {
+        final SlackEventNotificationConfig config = (SlackEventNotificationConfig) ctx.notificationConfig();
+        final String errorMessage = "SlackEventNotification send error for id : " + ctx.notificationId();
+
+        LOG.debug("SlackEventNotification backlog size in method execute is [{}]", config.backlogSize());
+
+
+        try {
+            SlackMessage slackMessage = createSlackMessage(ctx, config);
+            slackClient.send(slackMessage, config.webhookUrl());
+        } catch (TemporaryEventNotificationException exp) {
+            //scheduler needs to retry a TemporaryEventNotificationException
+            LOG.error(errorMessage, exp.toString());
```

Comment:
The `errorMessage` message template doesn't include a `{}` placeholder for the argument you are passing to `LOG.error`. That means the logged error message will not include the argument value.

When you look at the exception handler in `EventNotificationExecutionJob` you will see that it already includes the notification ID and title in the wrapping `JobExecutionException`. (which will then be logged by the `JobExecutionEngine`. That means you don't really need to log this in the exception handling code in this method.

Reply:
the argument value has been appended preemptively like below and hence doesn't need a `{}` placeholder.
`final String errorMessage = "SlackEventNotification execute error for id : " + ctx.notificationId();`

Reply:
But then you don't need to pass `exp.toString()` because it will just be ignored.

Reply:
i was trying to repurpose the errormessage across `log.error` and while constructing the exception and missed the idea of Log.error 2nd argment is a filler for the 1st argument.